logic/src/commonTest/kotlin/com/wire/kalium/logic/data/connection/ConnectionRepositoryTest.kt

```diff
@@ -161,14 +162,115 @@ class ConnectionRepositoryTest {
             .wasInvoked(once)
     }
 
+    @Test
+    fun givenAConnectionRequestUpdate_WhenSendingAConnectionStatusValid_thenTheConnectionShouldBePersisted() = runTest {
+        // given
+        val userId = NetworkUserId("user_id", "domain_id")
+        given(connectionApi)
+            .suspendFunction(connectionApi::updateConnection)
+            .whenInvokedWith(eq(userId), eq(ConnectionStateDTO.ACCEPTED))
+            .then { _, _ -> NetworkResponse.Success(connection1, mapOf(), 200) }
+        given(conversationDAO)
+            .suspendFunction(conversationDAO::updateOrInsertOneOnOneMemberWithConnectionStatus)
+            .whenInvokedWith(eq(UserIDEntity(userId.value, userId.domain)), any(), any())
+            .then { _, _, _ -> return@then }
```

Comment:
We could reuse many of these test configuration boilerplate calls if we used the Arrangement pattern 😛😅 Also imho it would help to make these tests more readable. In any case not a blocker

Reply:
Totally agree! I'm going to create a side branch, to do it 👍